jupyterhub/proxy.py

```diff
@@ -300,7 +301,7 @@ def check_routes(self, user_dict, service_dict, routes=None):
             user = user_dict[orm_user]
             if user.running:
                 good_routes.add(user.proxy_spec)
-                if user.name not in user_routes:
+                if user.name not in user_routes and not user.proxy_pending:
```

Comment:
Almost. `user.running` will never be True if proxy_pending is True, so the proxy_spec won't get added to `good_routes`, and thus will get deleted as stale in the next stage.